packages/pn-pa-webapp/src/navigation/routes.tsx

```diff
@@ -8,31 +10,60 @@ import NewNotification from '../pages/NewNotification.page';
 import PrivacyTOSPage from "../pages/PrivacyTOS.page";
 
 import { PNRole } from '../models/user';
+import { RootState } from '../redux/store';
+import { useAppDispatch, useAppSelector } from '../redux/hooks';
+import { AUTH_ACTIONS, getOrganizationParty } from '../redux/auth/actions';
 import TermsOfService from '../pages/TermsOfService.page';
-import * as routes from './routes.const';
 import SessionGuard from './SessionGuard';
 import RouteGuard from './RouteGuard';
+import * as routes from './routes.const';
+
+function OrganizationPartyGuard() {
+  const loggedUser = useAppSelector((state: RootState) => state.userState.user);
+  const idOrganization = loggedUser.organization?.id;
+  const dispatch = useAppDispatch();
+  const { t } = useTranslation(['common']);
+
+  const fetchOrganizationParty = useCallback(() => {
+    if (idOrganization) {
+      void dispatch(getOrganizationParty(idOrganization));
+    }
+  }, [idOrganization]);
+
+  return (
+    <ApiErrorWrapper 
+      apiId={AUTH_ACTIONS.GET_ORGANIZATION_PARTY} 
+      reloadAction={() => fetchOrganizationParty()}
+      mainText={t('messages.error-in-initial-info-loading')}
+      mt={3}
+    >
+      <Outlet />
+    </ApiErrorWrapper>
+  );
+}
 
 function Router() {
   return (
     <Routes>
       <Route path="/" element={<SessionGuard />}>
-        {/* protected routes */}
-        <Route path="/"  element={<RouteGuard roles={[PNRole.ADMIN, PNRole.OPERATOR]} />}>
-          <Route path={routes.TOS} element={<TermsOfService />} />
-          <Route path={routes.DASHBOARD} element={<Dashboard />} />
-          <Route path={routes.DETTAGLIO_NOTIFICA} element={<NotificationDetail />} />
-          <Route path={routes.NUOVA_NOTIFICA} element={<NewNotification />} />
-          {/**
-           * Refers to PN-1741
-           * Commented out because beyond MVP scope
-           *
-           * LINKED TO:
-           * - "const BasicMenuItems" in packages/pn-pa-webapp/src/utils/role.utility.ts
-           * - BasicMenuItems in packages/pn-pa-webapp/src/utils/__TEST__/role.utilitytest.ts
-           *
-           * <Route path={routes.API_KEYS} element={<ApiKeys />} />
-           * */}
+        <Route path="/" element={<OrganizationPartyGuard />}>
```

Comment:
We can move the logic of OrganizationPartyGuard into App.tsx?

Reply:
I would prefer to keep OrganizationPartyGuard as a proper guard, since it does not affect all the routes, there are indeed public routes (currently PRIVACY_POLICY and TERMS_OF_SERVICE, but I imagine others could be added later) where this guard shouldn't apply. So I keep the code in its current form, at least until I receive a subsequent review.

Reply:
OK per mantenere il guard ma lo sposterei in file e scriverei il test relativo

Reply:
I did as requested, albeit I don't agree with the comment. The decision of keeping `OrganizationPartyGuard` inside `routes.tsx` was deliberate, as in my view, the component is (at least in principle) of no interest outside the `Router` definition, and the inclusion in the same file does not make the file too big or complex. In fact, `OrganizationPartyGuard` was not exported, and this decision conveys a meaning: the component is something particular to the stuff implemented in `routes.tsx`.  
Besides this, I don't agree to separate each React component in a separate file as a general criterion, the ability to include several definitions in the same class is an aspect of Javascript that actually attracts me in comparison with (the boring) Java.  
I remark that the case which raised the convenience of defining `OrganizationPartyGuard`, namely when the API call to getOrganizationParty yields an error, was already covered by a test in `routes.test.ts`.

In general, I guess that it could be convenient to think about the thin line between a design flaw and the personal preferences of each team member.